firedrake/randomfunctiongen.py

```diff
@@ -0,0 +1,345 @@
+                st += sp + ':arg V: :class:`.FunctionSpace`\n\n'
+                st += sp + ':returns: :class:`.Function`\n\n'
+                st += sp + s.replace('(', '(*').replace(')', '*)') + '\n\n'
+            elif 'size=None' in s:
+                name = s[:s.find('(')]
+                st += sp + name + '(*args, **kwargs)\n\n'
+                s = '*' + name + '* ' + s[len(name):]
+                st += sp + s.replace('(', '(*V, ').replace(', size=None', '').replace(')', '*)') + '\n\n'
+                st += sp + 'Generate a :class:`.Function` f = Function(V), randomise it by calling the original method *' + name + '* (...) with given arguments, and return f.\n\n'
+                st += sp + ':arg V: :class:`.FunctionSpace`\n\n'
+                st += sp + ':returns: :class:`.Function`\n\n'
+                st += sp + "The original documentation is found at `<https://bashtage.github.io/randomgen/generated/randomgen.legacy.legacy.LegacyGenerator." + name + ".html>`__, which is reproduced below with appropriate changes.\n\n"
+                st += sp + s.replace('(', '(*').replace(')', '*)') + '\n\n'
+            elif '.. math::' in s:
+                st += '\n' + sp + s + '\n\n'
+            else:
+                st += sp + s + '\n'
+
+        return st
+
+    if module_attr == 'RandomGenerator':
+
+        _Base = getattr(randomgen, module_attr)
+
+        _dict = {}
+
+        _dict["__doc__"] = ("\n"
+                            "    Container for the Basic Random Number Generators.\n"
+                            "\n"
+                            "    Users can pass to many of the available distribution methods\n"
+                            "    a :class:`.FunctionSpace` as the first argument to obtain a randomised :class:`.Function`.\n"
+                            "\n"
+                            "    .. note ::\n"
+                            "        FunctionSpace, V, has to be passed as\n"
+                            "        the first argument.\n"
+                            "\n"
+                            "    **Example**::\n"
+                            "\n"
+                            "        from firedrake import *\n"
+                            "        mesh = UnitSquareMesh(2,2)\n"
+                            "        V = FunctionSpace(mesh, 'CG', 1)\n"
+                            "        pcg = PCG64(seed=123456789)\n"
+                            "        rg = RandomGenerator(pcg)\n"
+                            "        f_beta = rg.beta(V, 1.0, 2.0)\n"
+                            "        print(f_beta.dat.data)\n"
+                            "        # produces:\n"
+                            "        # [0.56462514 0.11585311 0.01247943 0.398984 0.19097059 0.5446709 0.1078666 0.2178807 0.64848515]\n"
+                            "\n")
+
```

Comment:
Documentation looks mostly good, but Sphinx finds this one only with Python 3.7+, probably because, with < 3.7, this class has to be wrapped with the "Wrapper" class at the very end of this file.  Shall we try to fix this?

Reply:
I think we can live with sphinx failing to find something on an old Python.